pkg/helpers/helpers.go

```diff
@@ -887,11 +916,126 @@ func FeatureGatesArgs(featureGates []operatorapiv1.FeatureGate, component string
 func isValidFeatureGate(feature string, component string) bool {
-	if featureGates, ok := FeatureGatesMap[component]; ok {
+	if featureGates, ok := featureGatesMap[component]; ok {
 		if _, ok := featureGates[featuregate.Feature(feature)]; ok {
 			return true
 		}
 	}
 
 	return false
 }
+
+func defaultNilFeatureGates(componentKey string) []string {
+	return defaultNilFeatureGatesMap[componentKey]
+}
+
+func getComponentKey(operatorType string, componentName string) string {
+	var componentKey string
+	switch componentName {
+	case componentNameRegistration:
+		if operatorType == OperatorTypeClusterManager {
+			componentKey = componentKeyHubRegistration
+		}
+		if operatorType == OperatorTypeKlusterlet {
+			componentKey = componentKeySpokeRegistration
+		}
+	case componentNameWork:
+		if operatorType == OperatorTypeClusterManager {
+			componentKey = componentKeyHubWork
+		}
+		if operatorType == OperatorTypeKlusterlet {
+			componentKey = componentKeySpokeWork
+		}
+	}
+
+	return componentKey
+}
+
+func getConditionType(componentName string) string {
+	var conditionType string
+	switch componentName {
+	case componentNameRegistration:
+		conditionType = RegistrationFeatureGatesValid
+	case componentNameWork:
+		conditionType = WorkFeatureGatesValid
+	}
+
+	return conditionType
+}
+
+// CheckRegistrationFeatureGates check if feature gates are valid for RegistrationConfiguration, return
+// the valid feature gates, the condition and if there is error, this func will not update the condition
+// to the status of the cluster manager or klusterlet
+func CheckRegistrationFeatureGates(operatorType string,
+	configuration *operatorapiv1.RegistrationConfiguration) ([]string, metav1.Condition, error) {
+	if operatorType != OperatorTypeClusterManager && operatorType != OperatorTypeKlusterlet {
+		return nil, metav1.Condition{}, fmt.Errorf("not supported operatorType: %s", operatorType)
+	}
+
+	if configuration == nil {
+		return checkNilFeatureGates(operatorType, componentNameRegistration)
+	}
+	return checkFeatureGates(operatorType, componentNameRegistration, configuration.FeatureGates)
+}
+
+// CheckWorkFeatureGates check if feature gates are valid for WorkConfiguration, return
+// the valid feature gates, the condition and if there is error, this func will not
+// update the condition to the status of the cluster manager or klusterlet
+func CheckWorkFeatureGates(operatorType string,
+	configuration *operatorapiv1.WorkConfiguration) ([]string, metav1.Condition, error) {
+	if operatorType != OperatorTypeClusterManager && operatorType != OperatorTypeKlusterlet {
+		return nil, metav1.Condition{}, fmt.Errorf("not supported operatorType: %s", operatorType)
+	}
+
+	if configuration == nil {
+		return checkNilFeatureGates(operatorType, componentNameWork)
+	}
+	return checkFeatureGates(operatorType, componentNameWork, configuration.FeatureGates)
+}
+
+func checkNilFeatureGates(operatorType string, componentName string) ([]string, metav1.Condition, error) {
```

Comment:
not sure this condition reason is helpful. I think nil is a valid, and we just return common valid reason

Reply:
OK, so do you think the reason FeatureGatesEmpty is necessary？

Reply:
no it is not